packages/ui/src/components/flag/flag.native.tsx

```diff
@@ -0,0 +1,52 @@
+import { StyleSheet, View, ViewStyle } from 'react-native';
+
+import type { FlagAlignment } from './flag.shared';
+
+export interface FlagProps {
+  children: React.ReactNode;
+  align?: FlagAlignment;
+  img?: React.ReactNode;
+  reverse?: boolean;
+  viewBoxStyles?: ViewStyle;
+}
+
+const getFlagAlignment = (align: FlagAlignment) => {
+  switch (align) {
+    case 'top':
+      return 'flex-start';
+    case 'middle':
+      return 'center';
+    case 'bottom':
+      return 'flex-end';
+  }
+};
+
+export function Flag({
+  align = 'middle',
+  reverse = false,
+  img,
+  children,
+  viewBoxStyles = {},
+}: FlagProps) {
+  return (
+    <View
+      style={[
+        styles.container,
+        {
+          flexDirection: reverse ? 'row-reverse' : 'row',
+          alignItems: getFlagAlignment(align),
+        },
+        viewBoxStyles,
+      ]}
+    >
+      <View style={styles.container}>{img}</View>
```

Comment:
I think I'm not understanding `styles.container` here and how it is handling what the web version of `Flag` does? This code: https://github.com/leather-io/mono/blob/68bb5075d03fcb3e8b530df2ad457679952eb5ff/packages/ui/src/components/flag/flag.web.tsx#L50

Reply:
I didn't add any spacing yet as I wasn't sure. I can add it, no problem. 

Are we working off a visual design here? I can just add the same spacing as web `space.03`

Reply:
I've implemented this properly now to apply `margin`  appropriately on `reverse`

Reply:
Yeah, from what I understand, with these two components (Flag and ItemLayout) we are just trying to match the web version as they are layout components which should be consistent b/w web and mobile.

cc @mica000 if I'm not correct here?

Reply:
Thanks @fbwoolf . I'll just build them with the same setup and we can then update them when consuming them if needed